libr/debug/p/debug_native.c

```diff
@@ -1259,6 +1260,262 @@ static int r_debug_desc_native_open (const char *path) {
 	return 0;
 }
 
+#if __APPLE__
+#define DEFAULT_COREFILE_DEST "/cores/core.%u"
+#else
+#define DEFAULT_COREFILE_DEST "" // Must set for specific platform
+#endif
+
+static int r_debug_gcore_prepare_corefile (RDebug *dbg, const char *newcorefile) {
+#if __unix__ || (__APPLE__ && __MACH__) 
+	// XXX Due to platform specific implementation right now, only APPLE can get uid & gid.
+	char *corefile[MAXPATHLEN];
+	int corefile_fd;
+	RDebugInfo *info = r_debug_info (dbg, "");
+
+	/* If the newcorefile string is empty, setup a sane default based on system defaults */
+	if (strlen (newcorefile) == 0) {
+		snprintf (corefile, MAXPATHLEN, DEFAULT_COREFILE_DEST, dbg->pid);
+	} else {
+		strncpy (corefile, r_str_chop_ro (strdup (newcorefile)), MAXPATHLEN - 1);
+	}
+
+  	corefile_fd = open (corefile, O_RDWR | O_CREAT | O_EXCL, 0600);
+  	if (corefile_fd < 0) {
+    	perror ("open");
+    	return corefile_fd;
+  	}
+
+  	// Change ownership
+  	if (fchown (corefile_fd, info->uid, info->gid) != 0) {
+      	eprintf ("Failed to set core file ownership\n");
+      	return -1;
+  	}
+  	r_debug_info_free (info);
+
+  	return corefile_fd;
+#else
+#warning This system doesnt has coredump file generation
+#endif
+}
+
+static int get_bits () {
+#if __i386__ || __powerpc__
+	return R_SYS_BITS_32;
+#elif __x86_64__ || __mips__
```

Comment:
I took that code from the plugin definition
    #elif __mips__
    .bits = R_SYS_BITS_32 | R_SYS_BITS_64,
    .arch = "mips",
    .canstep = 0,